toolchain/diagnostics/diagnostic_emitter.h

```diff
@@ -79,6 +102,9 @@ struct Diagnostic {
 
   // Returns the formatted string. By default, this uses llvm::formatv.
   std::function<std::string(const Diagnostic&)> format_fn;
+
+  // Context for the diagnostic.
+  llvm::SmallVector<std::unique_ptr<Diagnostic>> context;
```

Comment:
Here and throughout -- what do you think about calling these "notes" as opposed to "context"?

For me, "notes" focuses better on what they are, while "context" is a (possibly) reason. There might be other reasons we want to attach notes beyond context.

Reply:
Switched to "notes"